### DISCUSSION TITLE
Call to token endpoint is failing to get Access Token for webapp

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#2125] [Q&A] [answered]
I am trying to get id_token and access token for webapp registered in developer.bentley.com , but getting error . I am able to get code from authorization endpoint but the call to token endpoint is failing.This is how I am making the call to token endpoint :

 const tokenResponse = await axios.post(
    "https://imsoidc.bentley.com/connect/token",
    {
     grant_type: "authorization_code",
     client_id: this.provider.client_id,
     code: this.provider.code,
     code_verifier: this.provider.code_verifier,
     redirect_uri: "http://localhost:5000/signin-oidc",
    },
    { headers: { "content-type": "application/json" } }
   );

**Comment by pmconne** [comment, marked as answer]
> getting error 

Supply the error please.

**Reply by Rahul-0108** [reply]
{
message : Request failed with status code 400,
error: invalid_request ,
error_description: grant_type is required
}

**Reply by aruniverse** [reply]
Take a look at: https://developer.bentley.com/apis/overview/authorization/
Try posting to `https://ims.bentley.com/connect/token`; grant_type = authorization_code looks right

**Reply by Rahul-0108** [reply]
Hi , getting the tokens from /token endpoint worked after following the documentaion : https://developer.bentley.com/apis/overview/authorization/ .Now , I am facing issue for logout functionality , I added a post logout redirect url while registering the webapp . There is no logout endpoint mentioned in the API documentaion . I assume it to be  https://ims.bentley.com/connect/endsession which I got from the https://ims.bentley.com/.well-known/openid-configuration url . 
This endpoint logs out the user but does not redirect to my post logout redirect uri . This is how I am making the call : https://ims.bentley.com/connect/endsession?redirect_uri=http://localhost:5000

**Reply by aruniverse** [reply]
Could you try `post_logout_redirect_uri` instead of `redirect_uri` ?
So it'd be https://ims.bentley.com/connect/endsession?post_logout_redirect_uri=http://localhost:5000

**Reply by Rahul-0108** [reply]
Hi , it worked, Thanks !!
